decoder/vaapidecoder_base.cpp

```diff
@@ -223,7 +224,10 @@ VaapiDecoderBase::setupVA(uint32_t numSurface, VAProfile profile)
 
     if (!m_externalAllocator) {
         //use internal allocator
-        m_allocator.reset(new VaapiSurfaceAllocator(m_display->getID()), unrefAllocator);
+        if (m_noNeedExtraSurface)
+            m_allocator.reset(new VaapiSurfaceAllocator(m_display->getID(), VaapiSurfaceAllocator::MINIMUM_EXTRA_BUFFER_SIZE), unrefAllocator);
```

Comment:
instead of use MINIMUM_EXTRA_BUFFER_SIZE, could we just let user set the extra buffer size?

Reply:
If so, user can not know how many surfaces are created.